### ISSUE TITLE
mod_admin_extra cannot be used with custom sm_db_type

### ISSUE THREAD

**Issue by who** [processone/ejabberd#977]
mod_admin_extra doesn't have a mechanism for respecting `sm_db_type`. Therefore, if you use a db other than mnesia (the default) for a session management database, many commands exposed by mod_admin_extra will fail.

Here's an example, where I am trying to ban a user with `sm_db_type: redis` in my config.

```
ejabberdctl ban_account user1 localhost reason
Problem 'exit {aborted,{no_exists,[session,{<<"user1">>,<<"localhost">>},4]}}' occurred executing the command.
Stacktrace: [{mnesia,abort,1,[{file,"mnesia.erl"},{line,310}]},
             {mnesia_index,dirty_read,3,
                           [{file,"mnesia_index.erl"},{line,176}]},
             {mod_admin_extra,get_sessions,2,
                              [{file,"src/mod_admin_extra.erl"},{line,696}]},
             {mod_admin_extra,get_resources,2,
                              [{file,"src/mod_admin_extra.erl"},{line,691}]},
             {mod_admin_extra,kick_sessions,3,
                              [{file,"src/mod_admin_extra.erl"},{line,684}]},
             {mod_admin_extra,ban_account,3,
                              [{file,"src/mod_admin_extra.erl"},{line,675}]},
             {ejabberd_commands,execute_command2,2,
                                [{file,"src/ejabberd_commands.erl"},
                                 {line,407}]},
             {ejabberd_ctl,call_command,3,
                           [{file,"src/ejabberd_ctl.erl"},{line,295}]}]

```

**Comment by badlop** [comment]
Apart from ban_account, what other commands have you confirmed fail when using your setup?

**Comment by who** [comment]
@badlop 
The ban_account method was the only command for which I have observed this bug, but I have not fully tested all of the available commands.

**Comment by badlop** [comment]
Ok, can you confirm the command now works correctly with odbc storage?

Looking at the source code, I guess those commands may also fail, can you test them and confirm they require improvement?
* connected_users_info
* num_active_users
* process_rosteritems
* users_session_info

**Comment by who** [comment]
@badlop 
I have verified that ban account is now fixed in ejabberd 16.03 when using a custom sm_db_type.
Thanks!
